### PR TITLE
planner: fix create binding panic when status.record-db-qp enable

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/29503

Problem Summary:
fix create binding panic when status.record-db-qp enable

### What is changed and how it works?
fix create binding panic when status.record-db-qp enable
### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: planner/core/preprocess.go [hunk 1]
```diff
@@ -453,6 +453,10 @@ func (p *preprocessor) checkBindGrammar(originNode, hintedNode ast.StmtNode, def
 			p.err = ddl.ErrOptOnTemporaryTable.GenWithStackByArgs("create binding")
 			return
 		}
+		tableInfo := tbl.Meta()
+		dbInfo, _ := p.ensureInfoSchema().SchemaByTable(tableInfo)
```
Comment (time-and-fate) [inline planner/core/preprocess.go:457]: Do we need to handle this error?

**Review by time-and-fate — APPROVED** [review]

File: planner/core/preprocess.go (hunk 1 above)
Comment (Reminiscent) [reply planner/core/preprocess.go:457]: The second return value is not an error. It means whether we can find the table. We have already checked this.

**Review by qw4990 — APPROVED** [review]

**Pull request merged**